### DISCUSSION TITLE
Procedure with interactive updates

### DISCUSSION THREAD

**Discussion by LongnoseRob** [pymeasure/pymeasure#480] [Q&A]
I am sorry if the title to this question might be a bit opaque.

Image we have an instrument or DUT of which I would like to read cyclic certain values e.g. temperatures in a certain intervall, e.g. once per second.
This could be done with a timed loop inside a procedure.

But how can I update a property of the same instument while the procedure is running from outside of the procedure e.g. like manual adjustments of properties of the instrument from a GUI?

Has anyone ever built some kind of scheduler with procedures?
Or am I overseeing something in the worker-procedure construct that would give interactive access to my instruments properties during the execution of a procedure?

If needed for the  discussion I could come up with a more detailled example.

Thanks for the input.

**Comment by bilderbuchi** [comment]
You could put this cyclic query loop into a separate thread, that you then start and are free to manually adjust things.
pyVISA is thread-safe, so if you are usinga pyVISAAdapter there should not be any command collisions.

**Reply by LongnoseRob** [reply]
@bilderbuchi Thanks for the recommendation with threading in combination with pyVisaAdapter.

**Reply by bilderbuchi** [reply]
Sure, no thing! I hope it works out for you.
